sunpy/net/tests/test_scraper.py

```diff
@@ -225,3 +227,53 @@ def test_yearly_overlap():
     # Should return a single file for 2013
     trange = TimeRange("2013-01-02", "2013-01-03")
     assert len(scraper.filelist(trange)) == 1
+
+
+@pytest.mark.parametrize(
+    ("error_code", "expected_number_calls", "error_message"),
+    [
+        (400, 1, "Bad Request"),
+        (403, 1, "Forbidden"),
+        (429, 6, "Too Many Requests"),
+        (504, 6, "Gateway Timeout"),
+    ],
+)
+def test_http_errors_with_enqueue_limit(error_code, expected_number_calls, error_message):
+    with patch("sunpy.net.scraper.urlopen") as mocked_urlopen:
+        mocked_urlopen.side_effect = HTTPError(
+            "http://example.com", error_code, error_message, {}, None
+        )
+        time_range = TimeRange("2012/3/4", "2012/3/4 02:00")
+        pattern = "http://proba2.oma.be/lyra/data/bsd/{{year:4d}}/{{month:2d}}/{{day:2d}}/{{}}_lev{{Level:1d}}_std.fits"
+        scraper = Scraper(pattern)
+        with pytest.raises(HTTPError, match=error_message) as excinfo:
+            scraper._httpfilelist(time_range)
+        assert excinfo.value.code == error_code  
+        assert mocked_urlopen.call_count == expected_number_calls
+
+
+def test_connection_error():
+    with patch('sunpy.net.scraper.urlopen') as mocked_urlopen:
+        mocked_urlopen.side_effect = URLError('connection error')
+        time = TimeRange('2012/3/4', '2012/3/4 02:00')
+        pattern = "http://proba2.oma.be/lyra/data/bsd/{{year:4d}}/{{month:2d}}/{{day:2d}}/{{}}_lev{{Level:1d}}_std.fits"
+        scraper = Scraper(pattern)
+        with pytest.raises(URLError, match = 'connection error'):
+            scraper._httpfilelist(time)
+
+
+def test_http_404_message(caplog):
+    log = logging.getLogger('sunpy')
+    log.setLevel(logging.DEBUG)
```

Comment:
Changing this broke the other tests. 

Can we change the log level via caplog instead?